scripts/notify-new-listings.js

```diff
@@ -0,0 +1,126 @@
+const pollWait = 250;
+
+// File to keep state across restarts. Stores the last seen event sequence ID,
+// which allows continuing polling from the correct place
+const stateFile = "./notify-new-listings.state";
+
+const queryEvents = (args) => {
+  var filter = {eventTypes: "listing/created,listing/updated"};
+  return integrationSdk.events.query(
+    {...args, ...filter}
+  );
+};
+
+const saveLastEventSequenceId = (sequenceId) => {
+  try {
+    fs.writeFileSync(stateFile, sequenceId);
+  } catch (err) {
+    throw err;
+  }
+};
+
+const loadLastEventSequenceId = () => {
+  try {
+    const data = fs.readFileSync(stateFile);
+    return parseInt(data, 10);
+  } catch (err) {
+    return undefined;
+  }
+};
+
+const analyzeEvent = (event) => {
+  if (event.attributes.resourceType == "listing") {
+    const listing = event.attributes.resource;
+    const listingId = event.attributes.resourceId.uuid;
+    const authorId = listing.relationships.author.data.id.uuid;
+    const listingState = listing.attributes.state;
+    const previousValues = event.attributes.previousValues;
+    if (event.attributes.eventType == "listing/created") {
+      switch(listingState) {
+      case "pendingApproval":
+        console.log(`A new listing is pending approval: listing ID ${listingId}, author ID: ${authorId}`);
+        break;
+      case "published":
+        console.log(`A new listing has been published: listing ID ${listingId}, author ID: ${authorId}`);
+        break;
+      };
+    } else if (event.attributes.eventType == "listing/updated") {
+      // listing is approved when the new state is published and the previous
+      // state was pendingApproval
+      if (listingState == "published"
+          && previousValues.attributes
+          && previousValues.attributes.state == "pendingApproval") {
+        console.log(`Listing has been approved by operator: listing ID ${listingId}, author ID: ${authorId}`);
+      } else if (listingState == "published"
+                 && previousValues.attributes
+                 && previousValues.attributes.state == "draft") {
+        console.log(`A new listing has been published: listing ID ${listingId}, author ID: ${authorId}`);
+      } else if (listingState == "pendingApproval"
+                 && previousValues.attributes
+                 && previousValues.attributes.state == "draft") {
+        console.log(`A new listing is pending approval: listing ID ${listingId}, author ID: ${authorId}`);
+      }
+    }
+  }
+};
+
+const pollLoop = (sequenceId) => {
+  var params = sequenceId ? {startAfterSequenceId: sequenceId} : {createdAtStart: startTime};
+  queryEvents(params)
+    .then(res => {
+      const events = res.data.data;
+      const lastEvent = events[events.length - 1];
+      const fullPage = events.length == res.data.meta.perPage;
+      const delay = fullPage? pollWait : pollIdleWait;
+      const lastSequenceId = lastEvent? lastEvent.attributes.sequenceId : sequenceId;
```

Comment:
Hmm. We don't seem to have Prettier in this repo. Might make sense to add it at some point.
```suggestion
      const lastSequenceId = lastEvent ? lastEvent.attributes.sequenceId : sequenceId;
```